pkg/provisioner/ironic/ironic.go

```diff
@@ -1042,6 +1095,15 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 	// Ironic has the settings it needs, see if it finds any issues
 	// with them.
 	switch nodes.ProvisionState(ironicNode.ProvisionState) {
+	case nodes.CleanFail:
+		if ironicNode.LastError == "" {
+			p.log.Info("failed but error message not available")
+			result.Dirty = true
+		} else {
+			p.log.Info("found error", "msg", ironicNode.LastError)
```

Comment:
The caller should log the error message, so I don't think we need to do it here.